diff -urN fontconfig-2.12.0.orig/conf.d/Makefile.am fontconfig-2.12.0/conf.d/Makefile.am
--- fontconfig-2.12.0.orig/conf.d/Makefile.am	2016-06-23 17:50:38.192585121 +0300
+++ fontconfig-2.12.0/conf.d/Makefile.am	2016-06-23 17:51:14.891650870 +0300
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf	\
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
diff -urN fontconfig-2.12.0.orig/conf.d/Makefile.in fontconfig-2.12.0/conf.d/Makefile.in
--- fontconfig-2.12.0.orig/conf.d/Makefile.in	2016-06-23 17:50:38.191585092 +0300
+++ fontconfig-2.12.0/conf.d/Makefile.in	2016-06-23 17:51:28.346041589 +0300
@@ -353,6 +353,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf	\
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
