diff -urN fontconfig-2.12.6.orig/conf.d/Makefile.am fontconfig-2.12.6/conf.d/Makefile.am
--- fontconfig-2.12.6.orig/conf.d/Makefile.am	2017-10-07 00:01:26.333563449 +0300
+++ fontconfig-2.12.6/conf.d/Makefile.am	2017-10-07 00:02:37.245555244 +0300
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-$(PREFERRED_HINTING).conf	\
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-metric-aliases.conf \
 	40-nonlatin.conf \
@@ -63,12 +64,12 @@
 	10-hinting-slight.conf		\
 	10-no-sub-pixel.conf		\
 	10-scale-bitmap-fonts.conf	\
+	11-lcdfilter-default.conf	\
 	10-sub-pixel-bgr.conf		\
 	10-sub-pixel-rgb.conf		\
 	10-sub-pixel-vbgr.conf		\
 	10-sub-pixel-vrgb.conf		\
 	10-unhinted.conf		\
-	11-lcdfilter-default.conf	\
 	11-lcdfilter-legacy.conf	\
 	11-lcdfilter-light.conf		\
 	20-unhint-small-vera.conf	\
