--- conf.d/65-nonlatin.conf	2008-05-04 08:06:25.000000000 +0530
+++ conf.d/65-nonlatin.conf.new	2008-09-26 17:38:38.000000000 +0530
@@ -49,9 +49,16 @@
 			<family>Lohit Bengali</family>
  			<family>Lohit Gujarati</family>
  			<family>Lohit Hindi</family>
+			<family>Lohit Marathi</family>
+			<family>Lohit Maithili</family>
+			<family>Lohit Kashmiri</family>
+			<family>Lohit Konkani</family>
+			<family>Lohit Nepali</family>
+			<family>Lohit Sindhi</family>
  			<family>Lohit Punjabi</family>
  			<family>Lohit Tamil</family>
- 			<family>Lohit Malayalam</family>
+ 			<family>Meera</family>
+			<family>Lohit Malayalam</family>
  			<family>Lohit Kannada</family>
  			<family>Lohit Telugu</family>
  			<family>Lohit Oriya</family>
@@ -111,8 +118,15 @@
 			<family>Lohit Bengali</family>
  			<family>Lohit Gujarati</family>
  			<family>Lohit Hindi</family>
+			<family>Lohit Marathi</family>
+			<family>Lohit Maithili</family>
+			<family>Lohit Kashmiri</family>
+			<family>Lohit Konkani</family>
+			<family>Lohit Nepali</family>
+			<family>Lohit Sindhi</family>
  			<family>Lohit Punjabi</family>
  			<family>Lohit Tamil</family>
+			<family>Meera</family>
  			<family>Lohit Malayalam</family>
  			<family>Lohit Kannada</family>
  			<family>Lohit Telugu</family>
@@ -153,13 +167,20 @@
 			<family>Lohit Bengali</family>
 			<family>Lohit Gujarati</family>
 			<family>Lohit Hindi</family>
+			<family>Lohit Marathi</family>
+			<family>Lohit Maithili</family>
+			<family>Lohit Kashmiri</family>
+			<family>Lohit Konkani</family>
+			<family>Lohit Nepali</family>
+			<family>Lohit Sindhi</family>
 			<family>Lohit Punjabi</family>
- 			<family>Lohit Tamil</family>
- 			<family>Lohit Malayalam</family>
- 			<family>Lohit Kannada</family>
- 			<family>Lohit Telugu</family>
- 			<family>Lohit Oriya</family>
- 			<family>LKLUG</family>
+			<family>Lohit Tamil</family>
+			<family>Meera</family>
+			<family>Lohit Malayalam</family>
+			<family>Lohit Kannada</family>
+			<family>Lohit Telugu</family>
+			<family>Lohit Oriya</family>
+			<family>LKLUG</family>
 		</prefer>
 	</alias>
 </fontconfig>
