diff -urp fontconfig-2.6.97.old/src/fcformat.c fontconfig-2.6.97/src/fcformat.c
--- fontconfig-2.6.97.old/src/fcformat.c	2009-02-16 11:14:54.000000000 +0000
+++ fontconfig-2.6.97/src/fcformat.c	2009-02-16 11:17:01.000000000 +0000
@@ -73,7 +73,7 @@
 
 #define FCMATCH_FORMAT	"%{file:-<unknown filename>|basename}: \"%{family[0]:-<unknown family>}\" \"%{style[0]:-<unknown style>}\""
 #define FCLIST_FORMAT	"%{?file{%{file}: }}%{=unparse}"
-#define PKGKIT_FORMAT	"%{[]family{Font(%{family|downcase|delete( )})\n}}%{[]lang{Font(:lang=%{lang|downcase|translate(_,-)})\n}}"
+#define PKGKIT_FORMAT	"%{[]family{font(%{family|downcase|delete( )})\n}}%{[]lang{font(:lang=%{lang|downcase|translate(_,-)})\n}}"
 
 
 static void

