From 0d47cfabd82cd9c02ec5711383f06599b0450ac0 Mon Sep 17 00:00:00 2001
From: Behdad Esfahbod <behdad@behdad.org>
Date: Tue, 7 Dec 2010 18:48:56 -0500
Subject: [PATCH] Bug 28958 - lang=en matches other langs

Patch from Akira TAGOH.
---
 src/fclang.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/fclang.c b/src/fclang.c
index f3b825a..1c78328 100644
--- a/src/fclang.c
+++ b/src/fclang.c
@@ -505,6 +505,7 @@ FcLangSetCompare (const FcLangSet *lsa, const FcLangSet *lsb)
 
 /*
  * Used in computing values -- mustn't allocate any storage
+ * XXX Not thread-safe
  */
 FcLangSet *
 FcLangSetPromote (const FcChar8 *lang)
@@ -515,6 +516,7 @@ FcLangSetPromote (const FcChar8 *lang)
     int			id;
 
     memset (ls.map, '\0', sizeof (ls.map));
+    ls.map_size = NUM_LANG_SET_MAP;
     ls.extra = 0;
     id = FcLangSetIndex (lang);
     if (id > 0)
-- 
1.7.11.4

